swo/mpt/extensions/runtime/events/utils.py

```diff
@@ -31,5 +44,65 @@ def setup_client():
         status_forcelist=[500, 502, 503, 504],
     )
     session.mount("http://", HTTPAdapter(max_retries=retries))
-    session.headers.update({"Authorization": f"Bearer {settings.MPT_API_TOKEN}"})
+    session.headers.update(
+        {"Authorization": f"Bearer {settings.MPT_API_TOKEN}", "User-Agent": "swo-extensions/1.0"}
+    )
     return session
+
+
+def _response_hook(span, request, response):
+    if not response.ok:
+        span.set_attribute("http.error", response.content)
+
+
+def instrument_logging():
+    resource = Resource(
+        attributes={
+            "service.name": settings.SERVICE_NAME,
+        }
+    )
+
+    if settings.USE_APPLICATIONINSIGHTS:
+        exporter = AzureMonitorTraceExporter(
+            connection_string=settings.APPLICATIONINSIGHTS_CONNECTION_STRING
+        )
+    else:
+        exporter = ConsoleSpanExporter()
+
+    trace_provider = TracerProvider(resource=resource)
+    trace_provider.add_span_processor(BatchSpanProcessor(exporter))
+    trace.set_tracer_provider(trace_provider)
+
+    DjangoInstrumentor().instrument()
+    RequestsInstrumentor().instrument(response_hook=_response_hook)
+    LoggingInstrumentor().instrument()
+
+
+def wrap_for_trace(func, event_type):
+    @wraps(func)
+    def wrapper(client, event):
+        tracer = trace.get_tracer(event_type)
+
+        match event_type:
+            case "orders":
+                object_id = event.data["id"]
```

Comment:
The Event object already has an id field so I think this logic can be removed.
IMO who is responsible to set the right unique id is the event producer (now the poller, then the pusher).
We should try to let this wrapper as general as we can.

Reply:
Cool!